test/integration/bucket_cleaner.py

```diff
@@ -42,27 +49,41 @@ def _should_remove_bucket(self, bucket: Bucket):
         return False
 
     def cleanup_buckets(self):
-        b2_api, _ = authorize((self.b2_application_key_id, self.b2_application_key))
-        buckets = b2_api.list_buckets()
+        buckets = self.b2_api.list_buckets()
         for bucket in buckets:
-            if not self._should_remove_bucket(bucket):
-                print('Skipping bucket removal:', bucket.name)
-            else:
-                print('Trying to remove bucket:', bucket.name)
+            self.cleanup_bucket(bucket)
+
+    def cleanup_bucket(self, bucket: Bucket):
+        b2_api = self.b2_api
+        if not self._should_remove_bucket(bucket):
+            logger.info('Skipping bucket removal:', bucket.name)
+        else:
+            logger.info('Trying to remove bucket:', bucket.name)
+            files_leftover = False
+            try:
+                b2_api.delete_bucket(bucket)
+            except BadRequest:
+                logger.info('Bucket is not empty, removing files')
+                files_leftover = True
+
+            if files_leftover:
```

Comment:
```python
try:
    b2_api.delete_bucket()
    return
except BadRequest:
    logger.info('removing files')

# here actually remove files
```
^-- would reduce indention
```

Reply:
I have by design kept this indent so I don't pollute git history too much :D